core/state/statedb.go

```diff
@@ -684,10 +684,16 @@ func (db *DB) GetRefund() uint64 {
 // Finalise finalises the state by removing the db destructed objects
 // and clears the journal as well as the refunds.
 func (db *DB) Finalise(deleteEmptyObjects bool) {
+	db.finalise(deleteEmptyObjects, false)
+}
+
+func (db *DB) finalise(deleteEmptyObjects bool, skipValidators bool) {
 	// Commit validator changes in cache to stateObjects
 	// TODO: remove validator cache after commit
-	for addr, wrapper := range db.stateValidators {
-		db.UpdateValidatorWrapper(addr, wrapper)
+	if !skipValidators {
```

Comment:
This seems to be off. We don't have to ignore validatorWrapper change in the state DB. We only want to not store the validatorWrapper history. By adding this condition to ignore validatorWrapper change in state calculation, it's actually modifying the structure of the latest state trie (thus changing the state root).

Reply:
Thanks for explaining this to me today. I agree with your assessment, and I am working on a pruning mechanism that doesn't work on the latest block to avoid altering the state root; only on prior blocks. We don't expect this change to impact the state root of the old blocks, so nodes going for this strategy are eligible for participating in the syncing process of other nodes.